price_rule.go

```diff
@@ -84,6 +84,77 @@ type PriceRulesResource struct {
 	PriceRules []PriceRule `json:"price_rules"`
 }
 
+// SetPrerequisiteSubtotalRange sets or clears the subtotal range for which a cart must lie within to qualify for the price-rule
+func (pr *PriceRule) SetPrerequisiteSubtotalRange(greaterThanOrEqualTo *string) error {
+	if greaterThanOrEqualTo == nil {
+		pr.PrerequisiteSubtotalRange = nil
+	} else {
+		if !validateMoney(*greaterThanOrEqualTo) {
+			return fmt.Errorf("failed to parse value as Decimal, invalid value")
+		}
+
+		pr.PrerequisiteSubtotalRange = &prerequisiteSubtotalRange{
+			GreaterThanOrEqualTo: *greaterThanOrEqualTo,
+		}
+	}
+
+	return nil
+}
+
+// SetPrerequisiteQuantityRange sets or clears the quantity range for which a cart must lie within to qualify for the price-rule
+func (pr *PriceRule) SetPrerequisiteQuantityRange(greaterThanOrEqualTo *int) {
+	if greaterThanOrEqualTo == nil {
+		pr.PrerequisiteQuantityRange = nil
+	} else {
+		pr.PrerequisiteQuantityRange = &prerequisiteQuantityRange{
+			GreaterThanOrEqualTo: *greaterThanOrEqualTo,
+		}
+	}
+}
+
+// SetPrerequisiteShippingPriceRange sets or clears the shipping price range for which a cart must lie within to qualify for the price-rule
+func (pr *PriceRule) SetPrerequisiteShippingPriceRange(lessThanOrEqualTo *string) error {
+	if lessThanOrEqualTo == nil {
+		pr.PrerequisiteShippingPriceRange = nil
+	} else {
+		if !validateMoney(*lessThanOrEqualTo) {
+			return fmt.Errorf("failed to parse value as Decimal, invalid value")
+		}
+
+		pr.PrerequisiteShippingPriceRange = &prerequisiteShippingPriceRange{
+			LessThanOrEqualTo: *lessThanOrEqualTo,
+		}
+	}
+
+	return nil
+}
+
+// SetPrerequisiteToEntitlementQuantityRatio sets or clears the ratio between ordered items and entitled items (eg. buy X, get y free) for which a cart is eligible in the price-rule
+func (pr *PriceRule) SetPrerequisiteToEntitlementQuantityRatio(prerequisiteQuantity *int, entitledQuantity *int) {
+	if prerequisiteQuantity == nil && entitledQuantity == nil {
+		pr.PrerequisiteToEntitlementQuantityRatio = nil
+		return
+	}
+
+	var pQuant, eQuant int
+	if prerequisiteQuantity == nil {
+		pQuant = 0
```

Comment:
The default value of `pQuant` and `eQuant` are 0 so you could shorten the if-else checks to just check if != nil

Reply:
> Could the call still be done with a custom NewRequest? Not sure if this is the approach we want to take, but it was used in other cases instead of making changes to this repo.

Yes, we could take that approach as well, though it does create more tech debt for us each time we do it. If we cannot move forward with this approach (or the other suggested), then it will at least unblock our work.

Reply:
Addressed the if-statements mentioned above.